deltachat-ffi/src/lib.rs

```diff
@@ -275,12 +275,10 @@ pub unsafe extern "C" fn dc_is_open(context: *mut dc_context_t) -> libc::c_int {
         eprintln!("ignoring careless call to dc_is_open()");
         return 0;
     }
-    let ffi_context = &mut *context;
-    let inner_guard = ffi_context.inner.read().unwrap();
-    match *inner_guard {
-        Some(_) => 0,
-        None => 1,
-    }
+    let ffi_context = &*context;
+    ffi_context
+        .with_inner(|ctx| ctx.sql.is_open() as libc::c_int)
+        .unwrap_or(0)
```

Comment:
This works but the problem is that this will fire the ERROR callback with an error message if the context is closed.  Which I recently learned is wrong since ERROR must be shown to the user.  `.with_inner()` should probbly be using a WARNING instead.  But even so in this case this is entirely valid so no point in writing a spurious warning.

So instead you could look at doing the context uwrapping manually, like e.g. #607 does.

Reply:
UIs will only close the context when the login/app goes out of scope -- i don't think a spurious error-cb will cause problems.  if the code receiving the event then uses the context to try to do things it would -- but i think it's just taking the string from the error event and displaying it.  so i practise there is no real problem. We have several bugs and other things i'd rather like (us if possible) to focus on.